### ISSUE TITLE
Hubspot Sync with over 10,000 companies fails

### ISSUE THREAD

**Issue by NickStallman** [mautic/mautic#7549]
## Bug Description
Enabling the Hubspot integration on a fresh Mautic install has resulted in this every time the sync cron runs.
```Hubspot Error
Cannot fetch more than 10000 total records for this endpoint
```

This is due to this change Hubspot made in 2016.
https://integrate.hubspot.com/t/upcoming-changes-to-recent-endpoints-for-companies-and-deals/1121

To do the initial import, a different API endpoint is needed to prevent hitting that error.
This is from HubspotApi.php getCompanies() due to the v2/companies/recent/modified API call.

What would probably work best is additional code added to detect this error, then switch over to the companies/v2/companies/paged API in order to complete successfully.

| Q   | A
| --- | ---
| Mautic version | 2.15.1
| PHP version | 7.1.20
| Browser | 

### Steps to reproduce
1. Add more than 10,000 companies to Hubspot
2. Enable Mautic Hubspot Sync
 
### Log errors
```
[2019-05-21 21:53:12] mautic.ERROR: INTEGRATION ERROR: Hubspot - Cannot fetch more than 10000 total records for this endpoint [] []
```

**Comment by mundadashri** [comment]
We are also facing this issue. We need to get a quick fix for this on priority.
Please help.

**Comment by mundadashri** [comment]
It's almost a year that this bug was highlighted. Any update on this.
Please address this issue.

**Comment by mundadashri** [comment]
@NickStallman Were you able to find any workaround for this issue.
We are on self-hosted Mautic v2.16.2

I keep getting this every time the "mautic:integration:fetchleads --integration=Hubspot" cronjob runs.
![image](https://user-images.githubusercontent.com/53873545/81899130-c3943a00-95d7-11ea-90e7-9ad89340abff.png)

**Comment by RCheesley** [comment]
Hi there, as we are only providing fixes for Mautic 3.x going forward, can you confirm if this continues in Mautic 3.0.1?
Thanks!

**Comment by NickStallman** [comment]
If it hasn't been fixed, then this will certainly still be a problem. Hubspot requires different API calls to get beyond the 10,000 limit (get all data vs get recently modified data).

It's on my todo list to upgrade my Mautic instances to 3, but I haven't gotten around to it.

**Comment by RCheesley** [comment]
Ok, you could test it locally if you have the facility to do so - we have a few bugs relating to the HubSpot integration which we are hoping to get merged in 3.0.2 in case you can help us testing!

**Comment by NickStallman** [comment]
@RCheesley ok I'll do a test fresh install today to assist.

My original investigation a year ago should still be the correct solution to fixing this bug.
It's fundamentally just that v2/companies/recent/modified cannot be used to get more than 10,000 records. Instead you need to use companies/v2/companies/paged.
Once the Mautic database is in sync, the recent/modified API call will work fine.

**Comment by NickStallman** [comment]
@RCheesley rather annoyingly Mautic 3 requires PHP 7.2.21 to 7.3.99.
My servers have PHP 5.6, 7.1 and PHP 7.4 available.

I now remember why I haven't been able to try out version 3 and have to continue with version 2.

**Comment by RCheesley** [comment]
Ah, what a pain! Thanks for looking into it though! You could use a local instance, or we have mautibox.com but I'd recommend caution using that with an integration like HubSpot as others can access the instance. 

Thanks again for taking a look, we have a WIP PR for 7.4 support but it's got quite a bit to be done as we have to bump a bunch of dependencies before we can add support for 7.4.

**Comment by mabumusa1** [comment]
@NickStallman I am following up on this issue, did you manage to get it solved? if so please share with us the PR for review, I would love to get it merged into the next release of Mautic

**Comment by NickStallman** [comment]
@mabumusa1 no sorry, I can't get newer Mautic versions working due to PHP version limits and I don't have time to explore this right now. Hopefully time will free up in a month or two.

**Comment by mabumusa1** [comment]
@NickStallman thanks for the update, I will keep this bug open. hopefully, we will have you or other members in the community to get it fixed

**Comment by NickStallman** [comment]
This is not a stale issue.
